ec/ecresource/deploymentresource/elasticsearch/v2/elasticsearch_payload.go

```diff
@@ -160,11 +168,14 @@ func (es *ElasticsearchTF) payload(ctx context.Context, res *models.Elasticsearc
 		res.Plan.AutoscalingEnabled = ec.Bool(es.Autoscale.ValueBool())
 	}
 
-	res.Settings, ds = elasticsearchTrustAccountPayload(ctx, es.TrustAccount, res.Settings)
-	diags.Append(ds...)
+	// Only add trust settings to update payload if trust has changed
+	if state == nil || !es.TrustAccount.Equal(state.TrustAccount) || !es.TrustExternal.Equal(state.TrustExternal) {
```

Comment:
Is `state == nil` the bootstrapping case? In either case, does it make sense to write these trust values in that case vs just relying on the deployment defaults?

Reply:
Yep that's the creation case. But we can't just skip it, it could be that the user has already added trust configs when creating the deployment. If the user has not configured anything, the plan will be empty so it won't add any payloads either (But this is handled inside the payload functions)